mteb/models/bmretriever_models.py

```diff
@@ -0,0 +1,252 @@
+        ):
+            batch_texts = prompted_sentences[i : i + batch_size]
+
+            if eos_token:
+                batch_texts = [text + eos_token for text in batch_texts]
+
+                batch_dict = self.tokenizer(
+                    batch_texts,
+                    max_length=max_length - 1,  # for EOS token
+                    return_token_type_ids=False,
+                    return_attention_mask=False,
+                    padding=False,
+                    truncation=True,
+                )
+                batch_dict["input_ids"] = [
+                    ids + [self.tokenizer.eos_token_id]
+                    for ids in batch_dict["input_ids"]
+                ]
+
+                batch_dict = self.tokenizer.pad(
+                    batch_dict,
+                    padding=True,
+                    pad_to_multiple_of=8,
+                    return_attention_mask=True,
+                    return_tensors="pt",
+                )
+
+            batch_dict = {k: v.to(self.device) for k, v in batch_dict.items()}
+            outputs = self.model(**batch_dict)
+            embeddings = last_token_pool(
+                outputs.last_hidden_state, batch_dict["attention_mask"]
+            )
+            all_embeddings.append(embeddings.cpu().numpy())
+
+        return np.concatenate(all_embeddings, axis=0)
+
+
+# Sources summarized from the paper's Table 8
+BMRETRIEVER_TRAINING_DATA = {
+    "PubMed": ["train"],  # https://huggingface.co/datasets/MedRAG/pubmed
+    "raw_arxiv": ["train"],  # https://huggingface.co/datasets/mteb/raw_arxiv
+    "medical_meadow_cord19": [
+        "train"
+    ],  # https://huggingface.co/datasets/medalpa/medical_meadow_cord19
+    "textbooks": ["train"],  # https://huggingface.co/datasets/MedRAG/textbooks
+    "statpearls": ["train"],  # https://huggingface.co/datasets/MedRAG/statpearls
+    "LitCovid_BioCreative": [
+        "train"
+    ],  # https://huggingface.co/datasets/KushT/LitCovid_BioCreative
+    "S2ORC": ["train"],  # https://github.com/allenai/s2orc
+    "MSMARCO": [
+        "train"
+    ],  # https://huggingface.co/datasets/tevatro/msmarco-passage-corpus
+    "BMRetriever/biomed_retrieval_dataset": [
+        "train"
+    ],  # https://huggingface.co/datasets/BMRetriever/biomed_retrieval_dataset
+}
+
+BMRetriever_410M = ModelMeta(
+    loader=partial(
+        BMRetrieverWrapper,
+        model_name_or_path="BMRetriever/BMRetriever-410M",
+        instruction_template=instruction_template,
+        torch_dtype=torch.float32,
+    ),
+    name="BMRetriever/BMRetriever-410M",
+    languages=["eng-Latn"],
+    open_weights=True,
+    revision="e3569bfbcfe3a1bc48c142e11a7b0f38e86065a3",
+    release_date="2024-04-29",
+    embed_dim=1024,
+    n_parameters=353_822_720,
+    memory_usage_mb=1349,
+    max_tokens=2048,
+    license="mit",
+    reference="https://huggingface.co/BMRetriever/BMRetriever-410M",
+    similarity_fn_name="dot",
```

Comment:
I think model actually use `cosine` function
```suggestion
    similarity_fn_name="cosine",
```